internal/validation/mock_trace_validation.go

```diff
@@ -65,8 +66,10 @@ func validateTimeStamps(span *cloudtrace.Span) error {
 	return nil
 }
 
-func AddSpans(uploadedSpans map[string]*cloudtrace.Span, lock *sync.Mutex,
-	delay time.Duration, ctx context.Context, spans ...*cloudtrace.Span) error {
+// AddSpans adds the given spans to the list of uploaded spans as long as there are no duplicate names.
+// If a duplicate span name is detected, it will not be written, and an error is returned.
+func AddSpans(uploadedSpans *[]*cloudtrace.Span, uploadedSpanNames map[string]bool,
```

Comment:
The function signature is getting pretty long which suggests that validation.AddSpans might be doing too many things.

What do you think of moving responsibility for delays and synchronization to the caller

So instead of this:
```
err = validation.AddSpans(uploadedSpans, uploadedSpanNames, lock, delay, ctx, span1, span2);
```
we'd have
```
if err := delay(ctx, delay); err != nil {...}
lock.Lock()
if err := validation.AddSpans(uploadedSpans, uploadedSpanNames, span1, span2); err != nil {...}
lock.Unlock()
```

Reply:
by convention, context.Context should always be the first argument

Reply:
Yes that seems much better, thanks.